File: tests/fixtures/orm-models/two-stores/src/main/java/com/acme/audit/AuditEvent.java
package com.acme.audit;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/** An audit record, kept in the document store. */
@Document(collection = "audit_events")
public class AuditEvent {
    @Id
    private String id;
    private String action;
}

File: tests/fixtures/kotlin/spring-boot-kotlin/src/main/kotlin/com/acme/shop/audit/AuditEvent.kt
package com.acme.shop.audit

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.mongodb.repository.MongoRepository
import java.time.Instant

/** What happened to an order, kept for auditors. */
@Document(collection = "audit_log")
data class AuditEvent(
    @Id val id: String? = null,
    @Field("order_id") val orderId: Long = 0,
    val action: String = "",
    val at: Instant = Instant.EPOCH,
)

interface AuditRepository : MongoRepository<AuditEvent, String>
